File: Data Structures & Algorithms/max-area-of-island/submission-1.java
class Solution {

    private final int[][] dirs = {{-1,0},{1,0},{0,-1},{0,1}};

    public int maxAreaOfIsland(int[][] grid) {

        int ROWS = grid.length;
        int COLS = grid[0].length;
        int islands = 0;

        for ( int r = 0; r < ROWS; r++ ) {
            for ( int c = 0; c < COLS; c++) {
                if ( grid[r][c] == 1) {
                    int res = dfs(grid, r, c);
                    islands = Math.max(res, islands);
                }
            }
        }
        return islands;
    }

    private int dfs(int[][] grid, int r, int c) {

        // Base Case
        if ( r < 0 || c < 0 || r >= grid.length || c >= grid[0].length || grid[r][c] == 0) {
            return 0;
        }

        // Recursive Case
        grid[r][c] = 0;
        int res = 1;
        for ( int[] dir : dirs) {
            int nr = r + dir[0];
            int nc = c + dir[1];
            res += dfs(grid, nr, nc);
        }
        return res;
    }
}

File: Data Structures & Algorithms/combinations/submission-1.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        
        res  = []
        nums = [i for i in range(1,n+1)]

        def dfs(i,subset):

            # Base Case
            if len(subset) == k:
                res.append(subset.copy())
                return

            if i >= n:
                return

            # Recursive Case
            subset.append(nums[i])
            dfs(i+1,subset)
            subset.pop()
            dfs(i+1,subset)

        dfs(0,[])
        return res